### PR TITLE
A tool to build a self-contained source tarball

### PR DESCRIPTION
**Pull request by glebm**
The does some pruning of the dependencies' files to avoid a 100 MiB+ tarball.
Even with that, the final tarball size is 17 MiB with xz, 25 MiB with gzip.

Building from the resulting tarball does not require `smpq` (`devilutionx.mpq` is included).
The version number is also set in the tarball itself instead of relying on git (which does not work with tarballs).

See the file documentation at the top of `tools/make_src_dist.py` for more information.

This tarball will be produced on release via GitHub CI.
The GitHub workflow has been tested using https://github.com/nektos/act:

```shell
act -W .github/workflows/dist_src_release.yml -j make_src_dist
```

Refs #3412 /cc @jpalus

### CODE REVIEW

File: .github/workflows/dist_src_release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    - name: Build
+      working-directory: ${{github.workspace}}
+      run: tools/make_src_dist.py
+
+    - name: Upload-Package
+      if: ${{ !env.ACT }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: devilutionx.tar.xz
```
Comment (AJenbo) [inline .github/workflows/dist_src_release.yml:30]:
```suggestion
        name: devilutionx-src.tar.xz
```

**Review by AJenbo — COMMENTED** [review]
I wonder how stable tis will be and how generic the tarball is (will it build on android and windows?)

**Comment by glebm** [comment]
> will it build on android and windows?

Good point. I've removed `FETCHCONTENT_FULLY_DISCONNECTED`, so now this tarball will build on Android and Windows (although it will fetch some dependencies there).

As to how generic it is, this tarball should work everywhere except Android and Windows.

Stability-wise, if we change the dependencies that must be vendored, we'll have to update the Python script.

**Review by AJenbo — APPROVED** [review]
I'm fine having this as long as it is maintained :)

**Comment by glebm** [comment]
@AJenbo I maintain packages for 2 Linux distributions so no worries there, looking forward to simplifying those packages for the next release

**Comment by jpalus** [comment]
@glebm Thanks a lot for driving this! It really makes build process way more straightforward. Just a few minor nits from my side after using first such dist from 1.3.0 release:
1. It would nice if the source distribution followed standard practice of putting sources into single top level directory `<name>-<version>`
2. Related to above it would nice if source distribution would be versioned so ie `devilution-src-1.3.0.tar.xz` (might be worth considering for other artifacts too btw)
3. Note that source distribution attached to 1.3.0 release is not actually created from 1.3.0 tag as it seems to include #3386

**Comment by AJenbo** [comment]
2. this would prevent `latest` links on github from working

**Comment by AJenbo** [comment]
@Vitexus @muzena you might find this relevant for the build systems you worked on

**Comment by muzena** [comment]
@glebm 
I tried to generate tar archive with command:

`sudo act -W .github/workflows/src_dist_release.yml -j make_src_dist`

In terminal output it seems all good, in the end it says that tar archive is generated in path:

`/home/mint/aaaaaaaa/devilutionX/build-src-dist/devilutionx-1.3.0-ac152589.tar.xz`

But there's no exist  "build-src-dist" dir with devilutionx-1.3.0-ac152589.tar.xz archive.

**Comment by glebm** [comment]
@muzena That directory is in the Docker container, not on your local machine.

You can just run `tools/make_src_dist.py`

**Comment by muzena** [comment]
@glebm 
Is all dependencies are included or it still need to download some packages?
Like fmt, see [log file](https://launchpadlibrarian.net/571567061/buildlog_ubuntu-focal-amd64.devilutionx_1.3.0~focal_BUILDING.txt.gz)

**Comment by glebm** [comment]
@muzena All dependencies *that must be vendored* are included. For fmt you should just use the existing package

**Comment by AJenbo** [comment]
`sudo apt install libfmt-dev`

**Comment by Vitexus** [comment]
> `sudo apt install libfmt-dev`

https://github.com/Vitexus/devilutionX/blob/98aaec15f84f63ec9a8b2a81383d48baff934aaf/debian/control#L5

**Pull request merged**